classes/Validate.php

```diff
@@ -718,6 +718,18 @@ public static function isEan13($ean13)
         return !$ean13 || preg_match('/^[0-9]{0,13}$/', $ean13);
     }
 
+    /**
+     * Check for barcode validity (GTIN)
+     *
+     * @param $gtin
+     *
+     * @return bool
+     */
+    public static function isGtin($gtin): bool
```

Comment:
This should construct the value object and the value object should do the validation.

Reply:
I don't think I should change the way this value object is made. It is supposed to be a copy of the Ean13 Value Object but validated on a 14 lenght.

Reply:
This is legacy code, we never introduced the VO in this Validation class I don't see why this should change now

Reply:
> This is legacy code,

It maybe is a legacy class, but it is new code, a new method and since you introduced value objects for this, you better use them to avoid copy pasting.

Also, there's one golden rule, the legacy can call the new code, but not the opposite.

Reply:
The email is kind of an exception compared to all other validations because it's not a simple regexp But I get your argument, although I don't think we should construct the VO here because it doesn't have the same signature at all (it throws an exception not return a boolean)

But to factorize the regexp at least I guess here we can rely on the one from the VO